react/features/local-recording/recording/flac/flacEncodeWorker.js

```diff
@@ -0,0 +1,416 @@
+            this._sampleRate,
+
+            // Mono channel
+            1,
+            this._bitDepth,
+
+            FLAC_COMPRESSION_LEVEL,
+
+            // Pass 0 in becuase of unknown total samples,
+            0,
+
+            // checksum, FIXME: double-check whether this is necessary
+            true,
+
+            // Auto-determine block size (samples per frame)
+            0
+        );
+
+        if (this._encoderId === 0) {
+            throw new Error('Failed to create libflac encoder.');
+        }
+
+        // initialize the encoder
+        const initResult = Flac.init_encoder_stream(
+            this._encoderId,
+            this._onEncodedData.bind(this),
+            this._onMetadataAvailable.bind(this)
+        );
+
+        if (initResult !== 0) {
+            throw new Error('Failed to initalise libflac encoder.');
+        }
+
+        this._state = EncoderState.WORKING;
+    }
+
+    /**
+     * Receive and encode new data.
+     *
+     * @param {*} audioData - Raw audio data.
+     * @returns {void}
+     */
+    encode(audioData) {
+        if (this._state !== EncoderState.WORKING) {
+            throw new Error('Encoder is not ready or has finished.');
+        }
+
+        if (!Flac.isReady()) {
+            throw new Error('Flac not ready');
+        }
+        const bufferLength = audioData.length;
+
+        // convert to Uint32,
+        // appearantly libflac requires 32-bit signed integer input
+        // FIXME: why unsigned 32bit array?
+        const bufferI32 = new Int32Array(bufferLength);
+        const view = new DataView(bufferI32.buffer);
+        const volume = 1;
+        let index = 0;
+
+        for (let i = 0; i < bufferLength; i++) {
+            view.setInt32(index, audioData[i] * (0x7FFF * volume), true);
+            index += 4; // 4 bytes (32bit)
+        }
+
+        // pass it to libflac
+        const status = Flac.FLAC__stream_encoder_process_interleaved(
+            this._encoderId,
+            bufferI32,
+            bufferI32.length
+        );
+
+        if (status !== 1) {
+            // get error
+
+            const errorNo
+                = Flac.FLAC__stream_encoder_get_state(this._encoderId);
+
+            console.error('Error during encoding', FLAC_ERRORS[errorNo]);
+        }
+    }
+
+    /**
+     * Signals the termination of encoding.
+     *
+     * @returns {void}
+     */
+    finish() {
+        if (this._state === EncoderState.WORKING) {
+            this._state = EncoderState.FINISHED;
+
+            const status = Flac.FLAC__stream_encoder_finish(this._encoderId);
+
+            console.log('flac encoding finish: ', status);
+
+            // free up resources
+            Flac.FLAC__stream_encoder_delete(this._encoderId);
+
+            this._data = this._exportFlacBlob();
+        }
+    }
+
+    /**
+     * Gets the stats.
+     *
+     * @returns {Object}
+     */
+    getStats() {
+        return {
+            'samplesEncoded': this._bufferSize
```

Comment:
This seems to be a buffer size that is set once in the constructor and never updated. I would expect the samplesEncoded to get incremented as we encode more samples